pandas/plotting/_matplotlib/core.py

```diff
@@ -230,13 +238,13 @@ def __init__(
         self.legend_handles: list[Artist] = []
         self.legend_labels: list[Hashable] = []
 
-        self.logx = kwds.pop("logx", False)
-        self.logy = kwds.pop("logy", False)
-        self.loglog = kwds.pop("loglog", False)
-        self.label = kwds.pop("label", None)
-        self.style = kwds.pop("style", None)
-        self.mark_right = kwds.pop("mark_right", True)
-        self.stacked = kwds.pop("stacked", False)
+        self.logx = type(self)._validate_log_kwd("logx", logx)
```

Comment:
Could we use a `staticmethod` so we could avoid this odd `type(self).`?

Reply:
ive started using this pattern even with static/classmethods to make it easier to see when a method call is harmless state-wise